airbyte-webapp/src/components/connectorBuilder/Builder/BuilderOptional.tsx

```diff
@@ -0,0 +1,25 @@
+import { faAngleDown, faAngleRight } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+import React, { useState } from "react";
+import { FormattedMessage } from "react-intl";
+
+import styles from "./BuilderOptional.module.scss";
+
+export const BuilderOptional: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <div className={styles.wrapper}>
+      <button
+        onClick={() => {
+          setIsOpen(!isOpen);
```

Comment:
It isn't a huge deal, but when this is clicked it clears out any errors that are shown on the required fields.

To reproduce: select OAuth, click into Token refresh endpoint, click away (Required error is shown), click on Optional fields. Happens both when closing and when opening.

Ideally, the errors shown on the required fields would not go away just because optional fields are opened or closed.

Reply:
Formik is starting to annoy me... I played around a bit and noticed that if you type something into the field, then delete it again it will not remove the error on opening the collapsed section. So I just preset it with an empty string and now it works.